match/lib/match/storage/google_cloud_storage.rb

```diff
@@ -234,74 +235,120 @@ def ensure_keys_file_exists(google_cloud_keys_file, google_cloud_project_id)
-          UI.input("Confirm with enter")
+          UI.message("")
+          UI.input("Confirm with [Enter]")
         end
 
         UI.important("Please never add the #{DEFAULT_KEYS_FILE_NAME.cyan} file in version control.")
-        UI.important("Instead please add the file to your .gitignore")
-        UI.input("Confirm with enter")
+        UI.important("Instead please add the file to your `.gitignore` file")
+        UI.message("")
+        UI.input("Confirm with [Enter]")
 
         return DEFAULT_KEYS_FILE_NAME
       end
 
       def ensure_bucket_is_selected
-        # In case the user didn't provide a bucket name yet, they will
-        # be asked to provide one here
+        created_bucket = UI.confirm("Did you already create a Google Cloud Storage bucket?")
         while self.bucket_name.to_s.length == 0
-          # Have a nice selection of the available buckets here
-          # This can only happen after we went through auth of Google Cloud
-          available_bucket_identifiers = self.gc_storage.buckets.collect(&:id)
-          if available_bucket_identifiers.count > 0
-            @bucket_name = UI.select("What Google Cloud Storage bucket do you want to use? (you can define it using the `google_cloud_bucket_name` key)", available_bucket_identifiers)
-          else
-            UI.error("Looks like your Google Cloud account for the project ID '#{self.google_cloud_project_id}' doesn't")
-            UI.error("have any available storage buckets yet. Please visit the following URL")
+          unless created_bucket
+            UI.message("Create a bucket at the following URL:")
             UI.message("")
             UI.message("\t\thttps://console.cloud.google.com/storage/browser".cyan)
             UI.message("")
-            UI.message("and make sure to have the right project selected on top of the page")
-            UI.message("click on " + "Create Bucket".cyan + ", choose a name and confirm")
+            UI.message("Make sure to select the right project at the top of the page!")
+            UI.message("")
+            UI.message("\t\t 1. Click 'Create bucket'".cyan)
+            UI.message("\t\t 2. Enter a unique name".cyan)
+            UI.message("\t\t 3. Select a geographic location for your bucket".cyan)
+            UI.message("\t\t 4. Make sure the storage class is set to 'Standard'".cyan)
+            UI.message("\t\t 5. Click 'Create' to create the bucket".cyan)
             UI.message("")
-            UI.input("Once you're finished, please confirm with enter")
+            UI.input("Press [Enter] once you created a bucket")
+          end
+          bucket_name = UI.input("Enter the name of your bucket: ")
+
+          # Verify if the bucket exists
+          begin
+            bucket_exists = !self.gc_storage.bucket(bucket_name).nil?
+          rescue Google::Cloud::PermissionDeniedError
+            bucket_exists = true
```

Comment:
I'm a little unsure what behavior we actually want to do here 🤔 I tried running this to see what would happen when I _didn't_ create a bucket and I got a `Google::Cloud::PermissionDeniedError` so the wizard continued on with the permission check. Getting a permission denied makes sense because Google don't want to give out if a bucket exists or not to anyone so I'm just not sure if doing this check or handling this check like this is the correct approach.

I think would either need to:
1) Remove the exists check all together because its just going to give a permission failure if it doesn't
2) Keep the check but throw a warning saying something like "We couldn't find the bucket. Did you enter it correctly? If not, enter the correct name. If so, check the permissions."

Maybe something like that?

Reply:
It is actually possible to know if a bucket exists or not. 
I tested this with a non existing bucket name like `rickclephas-fastlane-match-test-test`.
This will only check if a bucket exists in Google Cloud, it doesn't have to be in the users project.

Reply:
So for clarification the bucket names are publicly visible and unique across GCS.
> Bucket names reside in a single Cloud Storage namespace. This mean that:
> - Every bucket name must be unique.
> - Bucket names are publicly visible.

https://cloud.google.com/storage/docs/naming

The bucket method will return `nil` if the bucket doesn't exists and throw a `PermissionDeniedError` if it does exists but the service account doesn't have access.

So users who enter a bucket name that does exists but is not theirs will pass the exists test but will fail the permission check.

When testing this make sure to use a name that you are sure doesn't exists. 
E.g. try with a domain you own (but doesn't have a bucket).

Reply:
Thanks for the clarification! I entered a weird bucket name to test this out but I guess I may have entered one that already existed and I just got really unlucky 😇 I will test again!